### PR TITLE
feat(JOML): migrate action event

### PR DESCRIPTION
**Pull request by pollend**
this migrates ActivateEvent/ActivationRequest/ActivationPrediction along with changes to get the game to run.

### CODE REVIEW

File: engine/src/main/java/org/terasology/logic/common/ActivateEvent.java [hunk 1]
```diff
@@ -84,16 +84,24 @@ public int getActivationId() {
 
     public Vector3f getTargetLocation() {
         LocationComponent loc = target.getComponent(LocationComponent.class);
-        if (loc != null&& !Float.isNaN(loc.getWorldPosition().x)) {
```
Comment (jdrueckert) [inline engine/src/main/java/org/terasology/logic/common/ActivateEvent.java:87]: Do we know why this originally only checked x for being finite?

File: engine/src/main/java/org/terasology/logic/common/ActivateEvent.java [hunk 2]
```diff
@@ -84,16 +84,24 @@ public int getActivationId() {
...
+            if (result.isFinite()) {
+                return result;
+            }
         }
         return null;
     }
 
     public Vector3f getInstigatorLocation() {
         LocationComponent loc = instigator.getComponent(LocationComponent.class);
-        if (loc != null&& !Float.isNaN(loc.getWorldPosition().x)) {
```
Comment (jdrueckert) [inline engine/src/main/java/org/terasology/logic/common/ActivateEvent.java:95]: see above

File: engine/src/main/java/org/terasology/logic/common/ActivateEvent.java [hunk 3]
```diff
@@ -84,16 +84,24 @@ public int getActivationId() {
...
-            return loc.getWorldPosition();
+        if (loc != null) {
+            Vector3f result = loc.getWorldPosition(new Vector3f());
+            if (result.isFinite()) {
+                return result;
+            }
+            result.set(0, 0, 0);
+            return result;
```
Comment (jdrueckert) [inline engine/src/main/java/org/terasology/logic/common/ActivateEvent.java:104]: Why do we return [0,0,0] vector here but `null` above in `getTargetLocation()`?

File: engine/src/main/java/org/terasology/logic/characters/events/ActivationPredicted.java [hunk 4]
```diff
@@ -76,16 +76,24 @@ public int getActivationId() {
 
     public Vector3f getTargetLocation() {
         LocationComponent loc = target.getComponent(LocationComponent.class);
-        if (loc != null&& !Float.isNaN(loc.getWorldPosition().x)) {
```
Comment (jdrueckert) [inline engine/src/main/java/org/terasology/logic/characters/events/ActivationPredicted.java:79]: Do we know why this originally only checked x for being finite?

File: engine/src/main/java/org/terasology/logic/characters/events/ActivationPredicted.java [hunk 5]
```diff
@@ -76,16 +76,24 @@ public int getActivationId() {
...
+            if (result.isFinite()) {
+                return result;
+            }
         }
         return null;
     }
 
     public Vector3f getInstigatorLocation() {
         LocationComponent loc = instigator.getComponent(LocationComponent.class);
-        if (loc != null&& !Float.isNaN(loc.getWorldPosition().x)) {
```
Comment (jdrueckert) [inline engine/src/main/java/org/terasology/logic/characters/events/ActivationPredicted.java:87]: see above

File: engine/src/main/java/org/terasology/logic/characters/events/ActivationPredicted.java [hunk 6]
```diff
@@ -76,16 +76,24 @@ public int getActivationId() {
...
-            return loc.getWorldPosition();
+        if (loc != null) {
+            Vector3f result = loc.getWorldPosition(new Vector3f());
+            if (result.isFinite()) {
+                return result;
+            }
+            result.set(0, 0, 0);
```
Comment (jdrueckert) [inline engine/src/main/java/org/terasology/logic/characters/events/ActivationPredicted.java:95]: Why do we return [0,0,0] vector here but `null` above in `getTargetLocation()`?

File: engine/src/main/java/org/terasology/logic/common/ActivateEvent.java (hunk 1 above)
Comment (pollend) [reply engine/src/main/java/org/terasology/logic/common/ActivateEvent.java:87]: An assumption that if x is nan then the whole vector is nan. Method from joml does not use this assumption.

File: engine/src/main/java/org/terasology/logic/common/ActivateEvent.java (hunk 3 above)
Comment (pollend) [reply engine/src/main/java/org/terasology/logic/common/ActivateEvent.java:104]: I have no idea why we do this. It's kind of strange. Don't want to think about it too much. We can look at it in another pass.

Comment (jdrueckert) [reply engine/src/main/java/org/terasology/logic/common/ActivateEvent.java:104]: Okay, different question: Do we need this at all if we have the `return new Vector3f()` at the end?
```
            result.set(0, 0, 0);
            return result;
```

Comment (jdrueckert) [reply engine/src/main/java/org/terasology/logic/common/ActivateEvent.java:104]: Yeah I see that, but isn't a cleared vector equal to an empty vector?
So the code path leads to the return of the empty vector (`return new Vector3f()` in l.106) anyway, so there's no need to explicitly clear and return at that point, is there?

Comment (pollend) [reply engine/src/main/java/org/terasology/logic/common/ActivateEvent.java:104]: I'm assuming there is some math that will add it or something and a nan vector will throw a wrench into things. Better to clear it. Basically the same as passing out a new vector.

Comment (jdrueckert) [reply engine/src/main/java/org/terasology/logic/common/ActivateEvent.java:104]: I see okay. Fair point.

**Review by jdrueckert — APPROVED** [review]

**Pull request merged**